syntax = "proto3";

package layoutinspector.proto;
option java_package = "com.android.tools.idea.layoutinspector.proto";
option java_outer_classname = "SkiaParser";

service SkiaParserService {
  rpc GetViewTree(GetViewTreeRequest) returns (GetViewTreeResponse) {}
}

// Version of GetViewTreeRequest:
// - : skp field only, from: AS 4.0
// 1 : known_ids added, from: AS 4.1

message GetViewTreeRequest {
  bytes skp = 1;
  int32 version = 2;             // See above for description
  repeated int64 known_ids = 3;  // Sorted list of known ids
}

message InspectorView {
  string id = 1;
  int64 timestamp = 2;
  string type = 3;
  int32 x = 4;
  int32 y = 5;
  int32 width = 6;
  int32 height = 7;
  bytes image = 8;

  repeated InspectorView children = 9;
}

// Version of GetViewTreeResponse:
// - : Include images for all render nodes found, from: AS 4.0
// 1 : Collect images only for known ids, from AS 4.1

message GetViewTreeResponse {
  InspectorView root = 1;
  int32 version = 2;
}
